interbotix_ros_xseries/interbotix_xs_sdk/src/gripper_calib.cpp

```diff
@@ -0,0 +1,147 @@
+     * @param names - list of joint names
+     * @return int  - index of the gripper name
+     */
+    int find_gripper_index(const std::vector<std::string>& names){
+        return std::find(names.begin(),names.end(),_gripper_name)-names.begin();
+    }
+
+    /**
+     * @brief Subscriber callback for performing gripper calibration
+     */
+    void calib_callback(const sensor_msgs::JointState &msg){
+        static int count = 0;
+        if(count%40==0){
+            int index = find_gripper_index(msg.name);
+            float curr_gripper_pos = msg.position[index];
+            ROS_INFO("I heard this position [%f]",msg.position[index]);
+            if(std::abs(curr_gripper_pos-prev_val)>0.0001){
+                gripper_command_msg.name = _gripper_name;
+                gripper_command_msg.cmd = -200;
+                pub.publish(gripper_command_msg);
+            }
+            else{
+                ROS_INFO("The error is 0");
+                calib_complete_srv(curr_gripper_pos,_gripper_name);
+                ros::shutdown();
+            }
+            prev_val = curr_gripper_pos;
+        }
+        count++;
+    }
+
+    /**
+     * @brief ROS Service Client that sends the calibration offset values to SDK
+     * @param min_position_offset - gripper offset value
+     * @param _gripper_name - gripper name
+     */
+    void calib_complete_srv(float min_position_offset, std::string _gripper_name){
+        gripper_calib_srv.request.gripper_name = _gripper_name;
+        gripper_calib_srv.request.offset = min_position_offset;
+        bool complete = client.call(gripper_calib_srv);
+        if(complete){
+            ROS_INFO("Gripper calibration finished successfully");
+        }
+        else{
+            ROS_ERROR("Failed to call service torque_enable");
+        }
+    }
+
+};
+
+
+/**
+ * @brief loads the motor config yaml file and returns a vector of gripper names for calibration
+ * @param success - Flag to check if the loading process was succesful
+ * @param robot_name - Name of the robot
+ * @param yaml_path - Path to the motor config yaml file.
+ * @return - [out] vector of gripper names.
+ */
+std::vector<std::string> load_calibration_config(bool& success, std::string& robot_name, std::string yaml_path){
+    YAML::Node yaml_node;
+    std::vector<std::string> calibration_joints;
+    try
+    {
+        yaml_node = YAML::LoadFile(yaml_path);
+    }
+    catch (YAML::BadFile &error)
+    {
+        ROS_FATAL("[xs_sdk] Calibration Config file was not found or has a bad format. Shutting down...");
+        ROS_FATAL("[xs_sdk] YAML Error: '%s'", error.what());
+        success = false;
+    }
+    if (yaml_node.IsNull())
+    {
+        ROS_FATAL("[xs_sdk] Calibration Config file was not found. Shutting down...");
+        success = false;
+    }
+    YAML::Node all_grippers = yaml_node["grippers"];
+    for(const auto gripper:all_grippers){
```

Comment:
Spaces between operators. Loop braces on next line
```suggestion
    for(const auto gripper : all_grippers)
    {
```